test/tooling.test.ts

```diff
@@ -90,6 +91,57 @@ it('should get completions and return completions', async () => {
   assert.ok(isObject(res.publicDeclarations));
 });
 
+it('can create static resource using application/json content type', async () => {
+  const request = createStaticResourceRequest(
+    fs.readFileSync('test/data/test.zip').toString('base64'),
+  );
+  const record = await conn.tooling.create('StaticResource', request);
+  assert.ok(record.success);
+});
+
+it('can create static resource using multipart/form-data content type with a buffer', async () => {
+  const request = createStaticResourceRequest(
+    fs.readFileSync('test/data/test.zip'),
+  );
+  const options = {
+    multipartFileFields: {
+      Body: {
+        contentType: 'application/zip',
+        filename: 'test.zip',
+      },
+    },
+  };
+  const record = await conn.tooling.create('StaticResource', request, options);
+  assert.ok(record.success);
+});
+
+it('can create static resource using multipart/form-data content type with a base64 encoded string', async () => {
+  const request = createStaticResourceRequest(
```

Comment:
question: wondering why we would need to base-64 encode the file, wouldn't the caller just pass the zip file (like in the previous test) ?

Reply:
While iterating on this I had some (incorrect) changes to  `request-helper.ts` that resulted in passing a Buffer to fail but passing a string to work, so I wanted to have tests for both scenarios.

Reply:
It passes a base64 encoded string that represents a zip file. The content type for a multipart "part" is not the same as the "Content-Type" HTTP header. Each "part" has to specify what its content type is. This is the equivalent of the curl command option you referenced in your research:

```
-F 'versionInfo=@package-version-info.zip;type=application/zip'
```

For binary content types (as opposed to, say, the json "part" we send for the non-file fields) you specify a content type for the _"part"_ and base64 encode the contents. In this case the `application/zip` content type for the `Body` part is used to tell the server what type of data will be present after decoding the base64 encoded string.

I tested that the StaticResource is saved and is identical when retrieved in both cases.

Reply:
Got it, thanks for clarifying that.